crates/network-scanner-net/src/async_io.rs

```diff
@@ -0,0 +1,152 @@
+use std::{
+    collections::HashMap,
+    num::NonZeroUsize,
+    sync::{
+        atomic::{AtomicBool, AtomicUsize},
+        Arc,
+    },
+};
+
+use anyhow::Context;
+use crossbeam::channel::Receiver;
+use parking_lot::Mutex;
+use polling::{Event, Events};
+use socket2::Socket;
+
+use crate::async_raw_socket::AsyncRawSocket;
+
+#[derive(Debug)]
+pub struct Socket2Runtime {
+    poller: polling::Poller,
+    next_socket_id: AtomicUsize,
+    is_terminated: AtomicBool,
+    map: Arc<Mutex<HashMap<usize, std::task::Waker>>>,
```

Comment:
suggestion: Maybe this doesn’t need to be wrapped in an `Arc`

Reply:
agree,updated.